pkg/tfgen/docs.go

```diff
@@ -1357,129 +1361,139 @@ func (g *Generator) convertExamplesInner(
 		contract.IgnoreError(err)
 	}
 
-	for _, section := range splitGroupLines(docs, "## ") {
-		if len(section) == 0 {
-			continue
+	// Find code fences in the presented doc
+	type tfBlockIndex struct {
+		start       int
+		end         int
+		headerStart int // a value of -1 indicates there's no header
+	}
+
+	var inCodeBlock bool
+	codeFence := "```"
+	h2 := "##"
+	h3 := "###"
```

Comment:
Are we not going to finds headers that are not h2 and h3s?

Reply:
It is possible but not very likely. The previous version of this code certainly only dealt in h2 and h3. We can make this more generic if we find that we are not "catching" some code blocks.

Reply:
hah, I found a better answer for you 💀 https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfgen/docs.go#L590